Python/Source/Select.cpp

```diff
@@ -0,0 +1,93 @@
+#include "Arithmetics.hpp"
+#include "Autobox.hpp"
+#include "Range.hpp"
+#include "Select.hpp"
+
+using namespace Rover;
+using namespace pybind11;
+
+namespace {
+  class PythonSelect {
+    public:
+      using Container = object;
+      using Selector = object;
+      using Type = object;
+
+      explicit PythonSelect(Container container)
+        : m_container(convert_container(std::move(container), false)),
+          m_selector(create_default_selector(m_container.attr("__len__")())) {}
+
+      PythonSelect(Container container, Selector selector)
+        : m_container(convert_container(std::move(container), true)),
+          m_selector(python_autobox<object>(std::move(selector))) {}
+
+      Type generate(Evaluator& evaluator) {
+        auto index = evaluator.evaluate(m_selector);
+        auto element = m_container.attr("__getitem__")(index);
+        auto generator = python_autobox<object>(element);
+        auto value = evaluator.evaluate(generator);
+        return value;
+      }
+
+    private:
+      static Box<object> create_default_selector(object&& size) {
+        auto begin = cast(0);
+        auto end = size - 1;
+        auto range = Details::PythonRange(std::move(begin), std::move(end),
+          cast(1));
+        auto selector = python_autobox<object>(cast(range,
+          return_value_policy::move));
+        return selector;
+      }
+
+      static Container convert_container(Container& container,
+          bool keep_dict) {
+        if(is_list(container)) {
```

Comment:
Based on how you perform the test, ```is_dict``` needs to be checked before ```is_list```. It's possible for a Python dict to use a number as its key.